### DISCUSSION TITLE
How can I link inside `\exception`?

### DISCUSSION THREAD

**Discussion by meator** [doxygen/doxygen#11296] [Q&A]
After experimenting a bit, I have decided to set `AUTOLINK_SUPPORT` to `NO` to reduce the amount of useless links Doxygen generates in my documentation. This has one unintended effect though: Doxygen no longer links exception objects in `\exception`.

With this input (`main.hpp`):

```cpp
/// @file

#include <stdexcept>

/**
 * @brief My exception class
 *
 * Detailed description.
 */
class MyError : public std::runtime_error
{
    using std::runtime_error::runtime_error;
};

/**
 * @brief Some function
 *
 * @exception MyError If function fails.
 */
void fie();
```

And with this `Doxyfile`:

```
# Doxyfile 1.12.0
INPUT = main.hpp
```

The `<exception-object>` of `fie()` gets correctly linked to `MyError`:

![image](https://github.com/user-attachments/assets/d0a80c96-9ce2-4c1c-bbab-4766bd750df6)

But when I set `AUTOLINK_SUPPORT=NO`:

```
# Doxyfile 1.12.0
INPUT = main.hpp
AUTOLINK_SUPPORT=NO
```

The link does not get generated:

![image](https://github.com/user-attachments/assets/6714a6f5-7850-4416-8eeb-cd42b535752c)

I have tried to link the `<exception-object>` manually, but was unable to do so:

```cpp
/// @file

#include <stdexcept>

/**
 * @brief My exception class
 *
 * Detailed description.
 */
class MyError : public std::runtime_error
{
    using std::runtime_error::runtime_error;
};

/**
 * @brief Some function
 *
 * @exception @ref MyError If function fails.
 */
void fie();
```

```
# Doxyfile 1.12.0
INPUT = main.hpp
AUTOLINK_SUPPORT=NO
```

This results in

```
warning: unexpected token TK_COMMAND_AT in comment block while parsing the argument of command exception
```

and in a malformed output:

![image](https://github.com/user-attachments/assets/9f0788c6-0284-4c75-8e2f-3c781f05feff)

How can I link inside `<exception-object>` when `AUTOLINK_SUPPORT` is set to `NO`?

**Info:**
- Doxygen version: latest (1.12.0)

**Comment by albert-github** [comment]
Unfortunately you didn't specify the doxygen version used nor a complete example.
I did try with the following code:
```
/// \file

/** @brief Some base function */
void fie0();

/**
 * @brief Some function
 *
 * @exception @ref fie0 If function fails.
 */
void fie();
```
and the setting:
```
QUIET = YES
```
the setting of `AUTOLINK_SUPPORT` here had no influence on the outcome.

When we look at the, important part of the, documentation of the `\exception` command:
> \exception `<exception-object>` `{ exception description }`
> 
> Starts an exception description for an exception object with name <exception-object>. Followed by a description
of the exception. The existence of the exception object is not checked. The text of the paragraph has no special
internal structure. All visual enhancement commands may be used inside the paragraph.

we see 2 parts: `<exception-object>`  and `{ exception description }` and I see that that you refer to the part `<exception-object>`.
From the documentation we learn that the `<exception-object>` is the name of the `<exception-object>` and that its existence is not checked and thus it is placed in the first column as "verbatim" text. 

You explicitly specified that you used `AUTOLINK_SUPORT = NO` as with `AUTOLINK_SUPPORT = YES` the line
```
@exception #fie0 If function fails.
```
results into a link to `fie0`

It might be a nice extension to the `\exception` (and `\throw`, `\throws`, `\retval` ...) command that the `<exception-object>` has the possibility to obey the `AUTOLINK_SUPPORT` / supports the `\ref` command for the `<exception-object>`.

**Reply by meator** [reply]
I have thought that my description of the issue was quite clear, but I've edited it after I read your response.

I can confirm that your suggestion of using

```
@exception #fie0 If function fails.
```

to link to `fie0` regardless of the value of `AUTOLINK_SUPPORT` does **not** work. No link is generated and the `#` character is ignored.
